dev/tests/mailchimp/tests/app/Ebizmarts/MailChimp/Model/ObserverTest.php

```diff
@@ -511,6 +512,157 @@ public function testNewOrder()
+        $customerLastname = 'Smith';
+        $isMarkedAsDeleted = 0;
+        $isMarkedAsDeleted = 1;
+        $type = Ebizmarts_MailChimp_Model_Config::IS_PRODUCT;
+
+
+        $itemMock = $this->getMockBuilder(Mage_Sales_Model_Order_Item::class)
+            ->disableOriginalConstructor()
+            ->setMethods(array('getProductType', 'getProductId'))
+            ->getMock();
+
+        $orderMock = $this->getMockBuilder(Mage_Sales_Model_Order::class)
+            ->disableOriginalConstructor()
+            ->setMethods(
+                array(
+                    'getStoreId', 'getCustomerEmail', 'getCustomerFirstname',
+                    'getCustomerLastname', 'getAllItems'
+                )
+            )
+            ->getMock();
+
+        $eventObserverMock = $this->getMockBuilder(Varien_Event_Observer::class)
+            ->disableOriginalConstructor()
+            ->setMethods(array('getEvent'))
+            ->getMock();
+
+        $eventMock = $this->getMockBuilder(Varien_Event::class)
+            ->disableOriginalConstructor()
+            ->setMethods(array('getOrder'))
+            ->getMock();
+
+        $observerMock = $this->getMockBuilder(Ebizmarts_MailChimp_Model_Observer::class)
+            ->disableOriginalConstructor()
+            ->setMethods(
+                array(
+                    'makeHelper', 'removeCampaignData', 'getMailchimpEcommerceSyncDataModel', 'isBundleItem',
+                    'isConfigurableItem'
+                )
+            )
+            ->getMock();
+
+        $helperMock = $this->getMockBuilder(Ebizmarts_MailChimp_Helper_Data::class)
+            ->disableOriginalConstructor()
+            ->setMethods(
+                array(
+                    'getMageApp', 'isEcomSyncDataEnabled',
+                    'isSubscriptionEnabled', 'loadListSubscriber',
+                    'saveEcommerceSyncData', 'getMCStoreId'
+                )
+            )
+            ->getMock();
+
+        $mageAppMock = $this->getMockBuilder(Mage_Core_Model_App::class)
+            ->disableOriginalConstructor()
+            ->setMethods(array('getRequest'))
+            ->getMock();
+
+        $requestMock = $this->getMockBuilder(Mage_Core_Controller_Request_Http::class)
+            ->disableOriginalConstructor()
+            ->setMethods(array('getPost'))
+            ->getMock();
+
+        $subscriberMock = $this->getMockBuilder(Mage_Newsletter_Model_Subscriber::class)
+            ->disableOriginalConstructor()
+            ->setMethods(array('getCustomerId', 'setSubscriberFirstname', 'setSubscriberLastname', 'subscribe'))
+            ->getMock();
+
+        $dataProductMock = $this->getMockBuilder(Ebizmarts_MailChimp_Model_Ecommercesyncdata::class)
+            ->setMethods(array('getMailchimpSyncDeleted', 'getEcommerceSyncDataItem', 'getMailchimpSyncModified'))
+            ->getMock();
+
+        $observerMock->expects($this->once())->method('makeHelper')->willReturn($helperMock);
```

Comment:
Please avoid creating all the mocks from scratch as the same exact code is in the test above.
Using auxiliar functions that are called by both tests should be more clean.